elastic/kibana
# [Response Ops][Alerting] SLIs Phase 2 - Histogram for task overdue by

Towards https://github.com/elastic/response-ops-team/issues/130

## Summary

This implements the third of 3 SLIs described in https://github.com/elastic/response-ops-team/issues/130 - the histogram for tracking task overdue time in seconds. We bucket up to 30 minutes. This implementation is slightly different than what was originally described for this SLI, which was number of overdue tasks. We changed to track a histogram of number of seconds overdue because this will provide a better metric against which to set an SLO.

## To Verify

Run Kibana and create some alerting rules. Navigate to https://localhost:5601/api/task_manager/metrics?reset=false and you should see that the new metric under `task_overdue.value`. There should be an overall task delay histogram and a histogram for each task type, with groupings for the alerting and action task types
